src/app/services/search.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable, EventEmitter } from '@angular/core';
+import { Movie } from '../models/movie.interface';
+import { TvShow } from '../models/tvshow.interface';
+import { Store } from '@ngrx/store';
+import { AppState } from '../store/state/app.state';
+import { selectMoviesList } from '../store/selectors/movies-page.selectors';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SearchService {
+  private filteredMovieArray: (Movie | TvShow)[];
+  private showedPage = 'movies';
+
+  public onQuickFilterSearch: EventEmitter<(Movie | TvShow)[]> = new EventEmitter();
+  public onSearchTurnOff: EventEmitter<boolean> = new EventEmitter();
+
+  constructor(private store: Store<AppState>) { }
+
+  public quickFilterSearch(searchInput: string) {
+    const regExp = new RegExp(searchInput, 'i');
+    if (this.showedPage === 'movies') {
+      this.store.select(selectMoviesList).subscribe(x => {
```

Comment:
You create new subscription in each type to search. Try to rethink this solution